scripts/sphinx_extensions/ncs_include/ncs_include.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# The code for this directive is based on the Include directive from
+# docutils, which is public-domain, stripped of lots of its original
+# options, and with added support for NCS documentation sets and
+# indentation # of the included rST.
+
+import os
+import os.path
+
+from docutils import io, statemachine
+from docutils.utils.error_reporting import SafeString, ErrorString
+from docutils.parsers.rst import directives
+from sphinx.util.docutils import SphinxDirective
+
+
+class NcsInclude(SphinxDirective):
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+    option_spec = {
+        'docset': directives.unchanged,
+        'indent': int,
+        'dedent': int,
+        'auto-dedent': directives.flag,
+        'tab-width': int,
+        'start-line': int,
+        'end-line': int,
+        'start-after': directives.unchanged_required,
+        'end-before': directives.unchanged_required,
+    }
+
+    def run(self):
+        if not isinstance(self.config.ncs_include_mapping, dict):
+            raise self.severe('Invalid "ncs_include_mapping" config')
+
+        docset = self.options.get('docset', None)
+        if docset is None:
+            raise self.severe('Need to supply a "docset" option')
```

Comment:
Since docset is a required option - can we define "nrf" as a docset as well, so we can take advantage of the indent/dedent options inside our main doc set as well?
Or even better, could the "local" doc set be the default?

Reply:
I both defined "nrf" and added code for using the local doc set when no "docset" was given.